lib/identity.rb

```diff
@@ -75,15 +75,15 @@ def tictac_dir
     end
 
     def public_key
-      @public_key ||= File.read(File.join(tictac_dir, 'self.ipfspub'))
+      @public_key ||= File.read(File.join(tictac_dir, "#{@keyname}.ipfspub"))
```

Comment:
referencing `@keyname` here means this method is broken if its used outside of the `setup` method

Maybe set `@keyname` in the initializer? You should have a different `Identity` object for each of the different keynames, since they correspond to different identities

Reply:
Yep that's exactly what is done in the 'add-tests' branch-- which is probably the one to review at this point.